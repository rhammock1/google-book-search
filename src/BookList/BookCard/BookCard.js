import React from 'react';

const BookCard = function(props) {
  return (
    <div className='book-card-container'>
      <h2>{props.book.title}</h2>
      <img src={props.book.cover} alt={props.book.title} />
      <p>Author: {props.book.author}</p>
      <p>Price: {props.book.price}</p>
      <p>{props.book.description}</p>
    </div>
    )
  

}

BookCard.defaultProps ={
  book: {
    title: ''
  }
}
export default BookCard;

