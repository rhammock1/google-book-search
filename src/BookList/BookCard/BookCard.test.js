import React, { createElement } from 'react';
import ReactDOM from 'react-dom';
import BookCard from './BookCard';
import renderer from 'react-test-renderer';


describe('Search component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookCard />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer.create(<BookCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})