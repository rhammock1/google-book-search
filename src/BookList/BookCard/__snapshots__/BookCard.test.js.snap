// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders the UI as expected 1`] = `
<div
  className="book-card-container"
>
  <h2>
    
  </h2>
  <img
    alt=""
  />
  <p>
    Author: 
  </p>
  <p>
    Price: 
  </p>
  <p />
</div>
`;
